test/model/message_list_test.dart

```diff
@@ -0,0 +1,241 @@
+import 'package:checks/checks.dart';
+import 'package:test/scaffolding.dart';
+import 'package:zulip/api/model/events.dart';
+import 'package:zulip/api/model/model.dart';
+import 'package:zulip/api/route/messages.dart';
+import 'package:zulip/model/message_list.dart';
+import 'package:zulip/model/narrow.dart';
+import 'package:zulip/model/store.dart';
+import '../api/fake_api.dart';
+import '../api/model/model_checks.dart';
+import '../model/binding.dart';
+import '../model/test_store.dart';
+import '../example_data.dart' as eg;
+
+const int userId = 1;
+const int streamId = 2;
+
+Future<PerAccountStore> setupStore(ZulipStream stream) async {
+  addTearDown(TestZulipBinding.instance.reset);
+
+  await TestZulipBinding.instance.globalStore.add(eg.selfAccount, eg.initialSnapshot());
+
+  final store = await TestZulipBinding.instance.globalStore.perAccount(eg.selfAccount.id);
+  store.addUser(eg.user(userId: userId));
+  store.addStream(stream);
+
+  return store;
+}
+
+Future<MessageListView> messageListViewWithMessages(List<Message> messages, PerAccountStore store, Narrow narrow) async {
+  final messageList = MessageListView.init(store: store, narrow: narrow);
+
+  final connection = store.connection as FakeApiConnection;
+
+  connection.prepare(json: GetMessagesResult(
+    anchor: messages.first.id,
+    foundNewest: true,
+    foundOldest: true,
+    foundAnchor: true,
+    historyLimited: false,
+    messages: messages,
+  ).toJson());
+
+  await messageList.fetch();
+
+  check(messageList.messages.length).equals(messages.length);
+
+  return messageList;
+}
+
+void main() async {
+  TestZulipBinding.ensureInitialized();
+
+  const narrow = StreamNarrow(streamId);
+  final stream = eg.stream(streamId: streamId);
```

Comment:
We can avoid needing a magic-number constant for the stream ID by putting these in the opposite order, to make the stream object available:
```suggestion
  final stream = eg.stream();
  final narrow = StreamNarrow(stream.streamId);
```